chains/nomad-ethereum/src/replica.rs

```diff
@@ -241,7 +241,7 @@ where
     }
 
     #[tracing::instrument(err)]
-    async fn prove(&self, proof: &Proof) -> Result<TxOutcome, ChainCommunicationError> {
+    async fn prove(&self, proof: &Proof<32>) -> Result<TxOutcome, ChainCommunicationError> {
```

Comment:
Maybe we can create `type NomadProof = Proof<TREE_DEPTH>` so we don't pass "32" everywhere? Not a big deal but awkward to have 32 floating all around the codebase